src/Common/REST/TEC/V1/Parameter_Types/Definition_Parameter.php

```diff
@@ -99,6 +100,53 @@ public function get_sanitizer(): Closure {
 		return $this->sanitizer ?? fn( $value ): array => $this->sanitize( $value );
 	}
 
+	/**
+	 * Filters the data.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data The data to filter.
+	 *
+	 * @return array The filtered data.
+	 *
+	 * @throws InvalidRestArgumentException If the data is invalid.
+	 */
+	public function filter( array $data = [] ): array {
+		$collections = $this->get_collections();
+
+		$filtered_data = [];
+
+		/** @var PropertiesCollection $collection */
+		foreach ( $collections as $collection ) {
+			/** @var Parameter $property */
+			foreach ( $collection as $property ) {
+				$param_name = $property->get_name();
+				$argument   = $this->get_name() ? $this->get_name() . '.' . $param_name : $param_name;
+
+				if ( $property->is_required() && ! isset( $data[ $param_name ] ) ) {
+					// translators: %s is the name of the property.
+					$exception = new InvalidRestArgumentException( sprintf( __( 'Property %s is required', 'the-events-calendar' ), $argument ) );
```

Comment:
We have this exact same Exception type and messaging in a few places. I think it would be better to add a static method to the `InvalidRestArgumentException` class that takes the argument name generates all of the details. Then call the static method to throw the exception.